Build failure: d2

### Steps To Reproduce

Steps to reproduce the behavior:

`nix build nixpkgs/nixos-unstable#d2`

### Build log

https://hydra.nixos.org/build/253624537/nixlog/1
```
...
Running phase: checkPhase
ok      oss.terrastruct.com/d2/d2ast    0.108s
ok      oss.terrastruct.com/d2/d2chaos  0.228s
ok      oss.terrastruct.com/d2/d2cli    0.042s
ok      oss.terrastruct.com/d2/d2compiler       0.436s
ok      oss.terrastruct.com/d2/d2exporter       0.876s
ok      oss.terrastruct.com/d2/d2format 0.016s
ok      oss.terrastruct.com/d2/d2graph  0.020s
--- FAIL: TestCompile (0.00s)
    --- FAIL: TestCompile/imports (0.20s)
        --- FAIL: TestCompile/imports/errors (0.00s)
            --- FAIL: TestCompile/imports/errors/escape (0.01s)
                import_test.go:235: 
                    --- a/build/ts_d2_diff944149770/exp
                    +++ b/build/ts_d2_diff944149770/got
                    @@ -1 +1 @@
                    -index.d2:1:1: failed to import "../x.d2": stat ../x.d2: invalid argument
                    \ No newline at end of file
                    +index.d2:1:1: failed to import "../x.d2": open ../x.d2: invalid argument
                    \ No newline at end of file
FAIL
FAIL    oss.terrastruct.com/d2/d2ir     0.253s
FAIL
```

### Notify maintainers

@dit7ya @kashw2

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7835,7 +7835,10 @@ with pkgs;
 
   choose = callPackage ../tools/text/choose { };
 
-  d2 = callPackage ../tools/text/d2 { };
+  d2 = callPackage ../tools/text/d2 {
+    # Failed tests
+    buildGoModule = buildGo121Module;
+  };
 
   ddev = callPackage ../applications/virtualization/ddev { };
 

